### PR TITLE
Remove `StreamMessageWriter` and replace it to `StreamWriter`

### PR DESCRIPTION
**Pull request by injae-kim**
**Related PR** https://github.com/line/armeria/pull/4696, **Issue** https://github.com/line/armeria/issues/4253
### Motivation:

> We neither provide an implemention that extends only `StreamWriter` nor public APIs that take `StreamWriter` as an input type.
> I’m not sure it is worth keeping both `StreamWriter` and `StreamMessageWriter`.
> It is likely that we can make `StreamWriter` extends `StreamMessage` and remove `StreamMessageWriter` for simplicity.

- See https://github.com/line/armeria/pull/4696#discussion_r1116919532
- We can remove `StreamMessageWriter` and replace it to `StreamWriter`

### Modifications:
- Remove `StreamMessageWriter` and replace it to `StreamWriter`
- Make `StreamWriter` extends `StreamMessage`

### Result:
- Now we can remove `StreamMessageWriter` and use `StreamWriter` instead for simplicity

### CODE REVIEW

**Review by ikhoon — APPROVED** [review]
Nice work! 👍💯

**Review by trustin — APPROVED** [review]
Very nice! 👍

File: core/src/main/java/com/linecorp/armeria/client/HttpResponseDecoder.java [hunk 1]
```diff
@@ -216,6 +215,37 @@ public boolean isOpen() {
             return delegate.isOpen();
         }
 
+        @Override
+        public boolean isEmpty() {
```
Comment (minwoox) [inline core/src/main/java/com/linecorp/armeria/client/HttpResponseDecoder.java:219]: How about throwing `UnsupportedOperationException` instead?

Comment (injae-kim) [reply core/src/main/java/com/linecorp/armeria/client/HttpResponseDecoder.java:219]: ✅ updated!

**Review by minwoox — APPROVED** [review]
Thaaaanks! 😄

**Review by jrhee17 — APPROVED** [review]
Thanks for the cleanup @injae-kim ! 🙇 👍 🚀

**Pull request merged**